flash/core/data/input_transform.py

```diff
@@ -406,17 +284,755 @@ def _create_collate_input_transform_processors(self) -> Tuple[Any]:
+        {
+            "input": ...,
+            "target": ...,
+            "metadata": ...,
+        }
+        """
+        return self._identity
+
+    def test_input_per_batch_transform_on_device(self) -> Callable:
+        """Defines the transform to be applied on the value associated with the "input" key of each single sample
+        on device for the testing stage."""
+        return self._identity
+
+    def test_target_per_batch_transform_on_device(self) -> Callable:
+        """Defines the transform to be applied on the value associated with the "target" key of each single sample
+        on device for the testing stage."""
+        return self._identity
+
+    def predict_per_batch_transform_on_device(self) -> Callable:
+        """Defines the transform to be applied on a batch of data on device for the predicting stage.
+
+        The input data of the transform would have the following form:
+        {
+            "input": ...,
+            "target": ...,
+            "metadata": ...,
+        }
+
+        You would need to use :class:`flash.core.data.transforms.ApplyToKeys` as follows:
+
+        Example:
+
+            from flash.core.data.transforms import ApplyToKeys
+
+            class MyInputTransform(InputTransform):
+
+                def per_batch_transform_on_device(self) -> Callable:
+
+                    return ApplyToKeys("input", my_func)
+        """
+        return self._identity
+
+    def predict_input_per_batch_transform_on_device(self) -> Callable:
+        """Defines the transform to be applied on the value associated with the "input" key of each single sample
+        on device for the predicting stage."""
+        return self._identity
+
+    def predict_target_per_batch_transform_on_device(self) -> Callable:
+        """Defines the transform to be applied on the value associated with the "target" key of each single sample
+        on device for the predicting stage."""
+        return self._identity
+
+    ###########
+    # COLLATE #
+    ###########
+
+    def train_collate(self) -> Callable:
+        """Defines the transform to be applied on a list of training sample to create a training batch."""
+        return default_collate
+
+    def val_collate(self) -> Callable:
+        """Defines the transform to be applied on a list of validating sample to create a validating batch."""
+        return default_collate
+
+    def test_collate(self) -> Callable:
+        """Defines the transform to be applied on a list of testing sample to create a testing batch."""
+        return default_collate
+
+    def predict_collate(self) -> Callable:
+        """Defines the transform to be applied on a list of predicting sample to create a predicting batch."""
+        return default_collate
+
+    def collate(self) -> Callable:
+        """Defines the transform to be applied on a list of sample to create a batch for all stages."""
+        return default_collate
+
+
+class LambdaInputTransform(InputTransform):
```

Comment:
Yes, I am not a huge fan of it, but at least it would make onboarding slightly simpler for user who just have a single callable and apply it on sample traditionally. This enables us to have a singly API internally to handle this onboarding path.

Reply:
Oh yeah, I agree. It's just that putting "Input" in the name feels wonky and misleading.